core/src/main/scala/scalafix/rewrite/Xor2Either.scala

```diff
@@ -13,16 +14,90 @@ case object Xor2Either extends Rewrite {
+    case class ReplaceType(original: Type,
+                           replacement: Type,
+                           newString: String)
+    val xorT: ReplaceType =
+      ReplaceType(t"cats.data.XorT", t"scala.util.EitherT", "EitherT")
+    val xor: ReplaceType =
+      ReplaceType(t"cats.data.Xor", t"scala.util.Either", "Either")
+    val xorLeft: ReplaceType =
+      ReplaceType(t"cats.data.Xor.Left", t"scala.util.Left", "Left")
+    val xorRight: ReplaceType =
+      ReplaceType(t"cats.data.Xor.Right", t"scala.util.Either.Right", "Right")
+
+    //experimenting to see if I can create lists of types to try to match.  No luck so far.
+    val replacementTypes: List[ReplaceType] =
+      List(xorT, xor, xorLeft, xorRight)
+
+    val termMap: Map[String, String] = Map(
+      "cats.data.Xor.Right.apply" -> "Right",
+      "cats.data.Xor.Left.apply" -> "Left"
+    )
+
     // NOTE. This approach is super inefficient, since we run semantic.desugar on
     // every case for every node in the tree. Ideally, we first match on the
     // syntax structure we want and then run semantic.desugar.
-    ast.collect {
-      case t @ DType(t"cats.data.Xor") =>
-        Patch(t.tokens.head, t.tokens.last, s"Either")
+    val tpes: Seq[(Type, ReplaceType)] = ast.collect {
+      case t @ DType(t"cats.data.XorT") => t -> xorT
+      case t @ DType(t"cats.data.Xor") => t -> xor
+      case t @ DType(t"cats.data.Xor.Left") => t -> xorLeft
+      case t @ DType(t"cats.data.Xor.Right") => t -> xorRight
+    }
+
+    //This is unsafe, come up with something better
+    val typeReplacements = tpes.map {
+      case (tree, rt) => Patch.replace(tree, rt.newString)
+    }
+
+    val terms: Map[String, Term] = ListMap(ast.collect {
       case t @ DTerm(q"cats.data.Xor.Right.apply[..$_]") =>
-        Patch(t.tokens.head, t.tokens.last, s"Right")
+        "cats.data.Xor.Right.apply" -> t
       case t @ DTerm(q"cats.data.Xor.Left.apply[..$_]") =>
-        Patch(t.tokens.head, t.tokens.last, s"Left")
+        "cats.data.Xor.Left.apply" -> t
+    }: _*)
+
+    val termReplacements = terms.map {
+      case (key, term) => Patch.replace(term, termMap(key))
+    }
+
+    val allImports = ast.collect {
+      case t @ q"import ..$importersnel" => t -> importersnel
     }
+
+    val firstImport = allImports.head._1
+    val firstImportFirstToken = firstImport.tokens.head
+    val tokenBeforeFirstImport =
+      ast.tokens.takeWhile(_ != firstImportFirstToken).lastOption
+
+    val partitionedImports = allImports.map(
+      kv =>
+        kv._1 -> kv._2
+          .flatMap(_.importees)
+          .partition(imp => List("Xor", "XorT").contains(imp.syntax)))
+
+    //If no importees to be kept, delete the whole importer
+    val removedImports = partitionedImports.collect {
+      case (t, (remove, keep)) if keep.isEmpty => Seq(Patch.delete(t))
+      case (t, (remove, keep)) => remove.map(Patch.delete)
+    }.flatten
```

Comment:
Imports are still an open question, and they will require semantic information to handle hairy cases like relative imports etc. We had a brief discussion on gitter https://gitter.im/scalacenter/scalafix?at=58526907058ca967374c48b3

I think that imports will eventually be handled by a custom import rewrite. That rewrite will probably take liberty to force a particular import style

```scala
import a.b
import a.c

// OR
import a.{b, c}
```

Users might need to run the import rewrite once before running other scalafix rewrites in order to minimize the diff. Similar to formatting.

I suggest you don't worry too much about imports for now. Just let them be.